compiler/rustc_trait_selection/src/traits/fulfill.rs

```diff
@@ -498,14 +499,23 @@ impl<'a, 'b, 'tcx> FulfillProcessor<'a, 'b, 'tcx> {
                         obligation.cause.span,
                     ) {
                         Ok(()) => ProcessResult::Changed(vec![]),
-                        Err(ErrorHandled::TooGeneric) => {
+                        Err(NotConstEvaluatable::MentionsInfer) => {
                             pending_obligation.stalled_on.clear();
                             pending_obligation.stalled_on.extend(
                                 substs.iter().filter_map(TyOrConstInferVar::maybe_from_generic_arg),
                             );
                             ProcessResult::Unchanged
                         }
-                        Err(e) => ProcessResult::Error(CodeSelectionError(ConstEvalFailure(e))),
+                        Err(NotConstEvaluatable::MentionsParam) => ProcessResult::Error(
+                            CodeSelectionError(SelectionError::NotConstEvaluatable(
+                                NotConstEvaluatable::MentionsParam,
+                            )),
+                        ),
+                        Err(NotConstEvaluatable::Error(e)) => {
+                            ProcessResult::Error(CodeSelectionError(
+                                SelectionError::NotConstEvaluatable(NotConstEvaluatable::Error(e)),
+                            ))
+                        }
```

Comment:
This is a bit pedantic, but since I needed a moment to figure out these two branches were just forwarding things, I think it improves the readability

```suggestion
                        Err(e @ NotConstEvaluatable::MentionsParam | NotConstEvaluatable::Error(_)) => ProcessResult::Error(
                            CodeSelectionError(SelectionError::NotConstEvaluatable(e)),
                        ),
```

Reply:
I like this change, done! :)